src/encoded/commands/regulome_search.py

```diff
@@ -14,73 +16,100 @@
+    @property
+    def atlas(self):
+        try:
+            return self._atlas
+        except AttributeError:
+            app = get_app(self.config_file, self.app_name)
+            # So that different process won't compete for one bigWig handle
+            instance_specific_bws = {
+                'IC_matched_max': pyBigWig.open(
+                    resource_filename(
+                        'encoded', '../../bigwig_files/IC_matched_max.bw'
+                    )
+                ),
+                'IC_max': pyBigWig.open(
+                    resource_filename(
+                        'encoded', '../../bigwig_files/IC_max.bw'
+                    )
+                ),
+            }
+            self._atlas = RegulomeAtlas(
+                app.registry[SNP_SEARCH_ES],
+                bw_signal_map=instance_specific_bws,
+            )
+            return self._atlas
+
+    def __call__(self, region_query):
+        motif_columns = (
+            '{chrom}\t'
+            '{start}\t'
+            '{end}\t'
+            '{motif_name}\t'
+            '{strand}\t'
+            '{hit_motif_start}\t'
+            '{hit_motif_end}\t'
+            '{query}\t'
+            '{query_start}\t'
+            '{query_end}'
+        )
         # Get coordinate for queried region
         region_query = region_query.strip()
         try:
-            chrom, start, end = get_coordinate(region_query, assembly, atlas)
+            chrom, start, end = get_coordinate(
+                region_query, self.assembly, self.atlas
+            )
         except ValueError:
-            logging.error('Invalid input: {}'.format(region_query))
-            continue
+            return 1, 'Invalid input: {}'.format(region_query)
         result = {
             'chrom': chrom,
             'start': start,
             'end': end,
         }
         try:
             all_hits = region_get_hits(
-                atlas,
-                assembly,
+                self.atlas,
+                self.assembly,
                 chrom,
                 start,
                 end,
-                peaks_too=return_peaks or matched_pwm_peak_bed_only
+                peaks_too=self.return_peaks or self.matched_pwm_peak_bed_only
             )
-            evidence = atlas.regulome_evidence(
+            evidence = self.atlas.regulome_evidence(
                 all_hits['datasets'], chrom, int(start), int(end)
             )
-            if not matched_pwm_peak_bed_only:
-                result['score'] = atlas.regulome_score(
+            if not self.matched_pwm_peak_bed_only:
+                result['score'] = self.atlas.regulome_score(
                     all_hits['datasets'], evidence
                 )
                 result['features'] = evidence_to_features(evidence)
         except Exception:
-            logging.error(
-                'Regulome search failed on {}:{}-{}'.format(chrom, start, end)
+            return 1, 'Regulome search failed on {}:{}-{}'.format(
```

Comment:
is 1 the correct return code here?

Reply:
It failed here; so I return 1 as failed.